solidity/contracts/TokenizedAccountNFT.sol

```diff
@@ -0,0 +1,122 @@
+ * normally with withdrawBalances().
+ */
+
+// Temporary Note: This is an ERC721 because of the greater control offered for individual tokens, since we only want
+// a single token for the contract. With ERC1155 it is impossible to figure out who the owner of a token is without
+// being given the owner's address. ERC721 exposes ownerOf(TokenID) so we can determine who the owner is.
+contract TokenizedAccountNFT is ERC721, OwnableApprovableERC721 {
+    event SuccessfulWithdrawal(address[] erc20s);
+    event TryRecover();
+    event SuccessfulRecovery(address[] erc20s, uint256[] amounts);
+    event ThresholdsChanged(address[] newErc20s, uint256[] newAmounts);
+
+    address[] private thresholdErc20s;
+    uint256[] private thresholdAmounts;
+
+    uint256 public constant AccountId = 1; // The Id for the only token controlled by this contract
+
+    // TODO: Figure out what our uri needs to be and where the json files should be hosted
+    constructor(string memory accountName)
+        ERC721(string.concat("TokenizedAccount:", accountName), string.concat("TA:", accountName)) {
+
+        _mint(msg.sender, AccountId);
+    }
+
+    function getThresholds() public virtual view returns (address[] memory, uint256[] memory) {
+        return (thresholdErc20s, thresholdAmounts);
+    }
+
+    /**
+     * @dev Updates the threshold values used by the x/microtx module which determine precisely how much
+     * of each token should be left in the TokenizedAccount's x/bank account.
+     *
+     * Any excess balances will accumulate here, and may be retrieved by the owner with
+     * the withdrawBalances(erc20s) method
+     *
+     * The ERC20s specified here have EVM addressses, querying the x/erc20 module will reveal their
+     * Cosmos Coin counterparts
+     *
+     * @notice this function is access controlled, only the owner or an approved msg.sender may call this function
+     */
+    function setThresholds(address[] calldata newErc20s, uint256[] calldata newAmounts) public virtual onlyOwnerOrApproved(AccountId) {
+        require(newErc20s.length == newAmounts.length, "threshold values must have the same length");
+        // Clear the thresholds before overwriting
+        delete thresholdErc20s;
+        delete thresholdAmounts;
+
+        for (uint i=0; i<newErc20s.length; i++) {
+            thresholdErc20s.push(newErc20s[i]);
+            thresholdAmounts.push(newAmounts[i]);
+        }
+        emit ThresholdsChanged(newErc20s, newAmounts);
+    }
+
+    /**
+     * @dev Withdraws ERC20 balances from this contract to the owner, throws on any erc20 transfer() failure
+     * Emits a {SuccessfulWithdrawal} event upon success.
+     *
+     * @notice This function is access controlled, only the owner or an approved msg.sender may call this function
+    */
+
+    // TODO: Should we even accept a destination, or just send to the owner of the AccountId token?
```

Comment:
I think having a withdraw and withdraw_to would be a good idea.

Reply:
Added